### DISCUSSION TITLE
Help. useScroll is not reactive

### DISCUSSION THREAD

**Discussion by unrevised6419** [vueuse/vueuse#2069] [Q&A] [answered]
I created this example that uses `useScroll` on the `document` that has more content than screen size.
I have used a `watchEffect` that prints `y` whenever `document` is scrolled, but nothing happens.
From the source code, I see that `y` should be `scrollTop` of the element.
If I added `onScroll` option it is called each time I scroll.

What I am doing wrong in this example? Or `useScroll` is not supposed to work like this?

[https://sfc.vuejs.org/#eNrtmMtOG0EQRX...](https://sfc.vuejs.org/#eNrtmMtOG0EQRX+lNBuMZGb2CEgihV12UXbeND01dpF+0Q8DsvzvuT0DxE4kvqB3/bh9+1adXR26byH0+8LddXeTdJSQKXEu4W7jiMQGHzMd6FllvbufJtaZjjRFb+kCjy7ORCXxTx29MR+Sr9DgdNA+Vm1VG67SV0hu/z5YjV4Xyy5fLqKT71arS7q9I+1d8oZ747er10vIboYlLXJik9kGozLPqW+W8EQ/8K0lCalYGr3xkZJkUpbzejaEPUqNpEYJkrS4LbERXCYe8YBYSrJ+pOqOx+K0jDIWl6lkMuoB9oRyZmsmq7ZOkTLyVFRPvzKxEwtvslIXe2yVXdNTkUTOpxzLSPzCUUtWWbyjYoyy2i/OVSRJ6k+zpQSIiRWCW2TySwH4Kvf0vVqqkpkkFiRZahVHkUPkHbuRIwrHwd6bEvAdIw4qJU6JSYsx7x1CQYWmshWVydVAFFTEpsSe7l80h8xoWBL0wGutWEOnS5BRwRVlOQrRywiU6GLtFD7VxQRV6yY/TaJF0ciJY721gF+R1AYJ2pHe+lpsP5McFpSNaCPaiDaijWgj2og2oo1oI9qINqKNaCPaiDaijWgjek60DghPhoLdulvGlFdWhf4xeYd556HKN28XadNd03xSzzC3rPtNt8s5pOthSJOuU9LH1Pu4HbDqaxax3HOyVw/RPyMdjDfd+t3jdPh5blZc+L3tAWU41XwBomIg/XDA5QDbPcerOAPh+Fmqf6T/Jau2x407dsc/ABz/mw==)

**Comment by curtgrimes** [comment, marked as answer]
I'm experiencing the same issue as you on the SFC Playground, but not when I run your same demo (with one tweak) on StackBlitz -- it works successfully on StackBlitz.

Demo: https://stackblitz.com/edit/vue3-script-setup-with-vite-ppshjm?file=src%2FApp.vue

```diff
<script setup>
  import { watchEffect } from 'vue';
  import { useScroll } from '@vueuse/core';

  let { y } = useScroll(document);

- watchEffect(() => console.log(y));
+ watchEffect(() => console.log(y.value)); // log out the value instead of the ref
</script>

<template>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
    non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
  ... more template
</template>
```

**Reply by unrevised6419** [reply]
Thanks, @curtgrimes. Then it seems there is something wrong with Playground 🤔 

Previously I have used just a watch but also didn't work

```ts
watch(y, v => console.log(v))
```

I start thinking that this may be something to do with how `vue-demi` in Playground works, that is used under `@vueuse/core` hood

**Reply by curtgrimes** [reply]
I agree -- I also tried `watch()` and experienced the same thing as you. I think there is a bug in here somewhere but probably not on VueUse's side.
